docs/source/peft_integration.md

```diff
@@ -1,144 +1,672 @@
+```bash
+pip install bitsandbytes
+```
 
-## How to use it?
+## Quick Start
 
-Simply declare a [`~peft.PeftConfig`] object in your script and pass it through `.from_pretrained` to load the TRL+PEFT model.
+All TRL trainers support PEFT through the `peft_config` argument. The simplest way to enable PEFT is using the command-line interface with the `--use_peft` flag:
+
+```bash
+python trl/scripts/sft.py \
+    --model_name_or_path Qwen/Qwen2-0.5B \
+    --dataset_name trl-lib/Capybara \
+    --use_peft \
+    --lora_r 32 \
+    --lora_alpha 16 \
+    --output_dir Qwen2-0.5B-SFT-LoRA
+```
+
+Alternatively, you can pass a PEFT config directly in your Python code:
 
 ```python
 from peft import LoraConfig
-from trl import AutoModelForCausalLMWithValueHead
+from trl import SFTConfig, SFTTrainer
 
-model_id = "edbeeching/gpt-neo-125M-imdb"
-lora_config = LoraConfig(
-    r=16,
-    lora_alpha=32,
+# Configure LoRA
+peft_config = LoraConfig(
+    r=32,
+    lora_alpha=16,
     lora_dropout=0.05,
     bias="none",
     task_type="CAUSAL_LM",
 )
 
-model = AutoModelForCausalLMWithValueHead.from_pretrained(
-    model_id, 
-    peft_config=lora_config,
+# Create trainer with PEFT
+trainer = SFTTrainer(
+    model=model,
+    args=training_args,
+    train_dataset=dataset,
+    peft_config=peft_config,
 )
 ```
 
-And if you want to load your model in 8bit precision:
+## PEFT with Different Trainers
+
+TRL's trainers support PEFT configurations for various training paradigms. Below are detailed examples for each major trainer.
+
+### Supervised Fine-Tuning (SFT)
+
+The `SFTTrainer` is used for supervised fine-tuning on instruction datasets.
+
+#### Full Training (No PEFT)
+
+```bash
+python trl/scripts/sft.py \
+    --model_name_or_path Qwen/Qwen2-0.5B \
+    --dataset_name trl-lib/Capybara \
+    --learning_rate 2.0e-5 \
+    --num_train_epochs 1 \
+    --per_device_train_batch_size 2 \
+    --gradient_accumulation_steps 8 \
+    --output_dir Qwen2-0.5B-SFT
+```
+
+#### With LoRA
+
+```bash
+python trl/scripts/sft.py \
+    --model_name_or_path Qwen/Qwen2-0.5B \
+    --dataset_name trl-lib/Capybara \
+    --learning_rate 2.0e-4 \
+    --num_train_epochs 1 \
+    --per_device_train_batch_size 2 \
+    --gradient_accumulation_steps 8 \
+    --use_peft \
+    --lora_r 32 \
+    --lora_alpha 16 \
+    --output_dir Qwen2-0.5B-SFT-LoRA
+```
+
+#### Python Example
 
 ```python
-pretrained_model = AutoModelForCausalLMWithValueHead.from_pretrained(
-    config.model_name, 
-    load_in_8bit=True,
-    peft_config=lora_config,
+from datasets import load_dataset
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from peft import LoraConfig
+from trl import SFTConfig, SFTTrainer
+
+# Load model and tokenizer
+model = AutoModelForCausalLM.from_pretrained("Qwen/Qwen2-0.5B")
+tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2-0.5B")
+
+# Load dataset
+dataset = load_dataset("trl-lib/Capybara", split="train")
+
+# Configure LoRA
+peft_config = LoraConfig(
```

Comment:
We actually have 3 different ways of adding the peft config to the trainer:

1. We give the model_name to the Trainer and the peft_config
2. We give the model instance and at the peft_config
3. We give the peft_model to the trainer directly, preparing it outside, without passing peft_config to the trainer.

We could add these details somewhere.